extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -0,0 +1,124 @@
+	private projectProvider: IProjectProvider | undefined;
+	private overviewTab: azdata.DashboardTab | undefined;
+
+	constructor(private workspaceService: IWorkspaceService, private treeItem: WorkspaceTreeItem) {
+	}
+
+	public async showDashboard(): Promise<void> {
+		const project = this.treeItem.element.project;
+		this.projectProvider = await this.workspaceService.getProjectProvider(vscode.Uri.file(project.projectFilePath));
+		if (!this.projectProvider) {
+			throw new Error(constants.ProviderNotFoundForProjectTypeError(project.projectFilePath));
+		}
+
+		await this.createDashboard(project.projectFileName);
+		await this.dashboard!.open();
+	}
+
+	private async createDashboard(title: string): Promise<void> {
+		this.dashboard = azdata.window.createModelViewDashboard(title, 'ProjectDashboard', { alwaysShowTabs: false });
+		this.dashboard.registerTabs(async (modelView: azdata.ModelView) => {
+			this.modelView = modelView;
+
+			this.overviewTab = {
+				title: '',
+				id: 'overview-tab',
+				content: this.createContainer(),
+				toolbar: this.createToolbarContainer()
+			};
+			return [
+				this.overviewTab
+			];
+		});
+	}
+
+	private createToolbarContainer(): azdata.ToolbarContainer {
+		const projectActions: (IProjectAction | IProjectActionGroup)[] = this.projectProvider!.projectActions;
+
+		// Add actions as buttons
+		const buttons: azdata.ToolbarComponent[] = [];
+
+		projectActions.forEach(action => {
+			if (this.isProjectAction(action)) {
+				let button = this.createButton(action);
+
+				buttons.push({ component: button });
+			} else {
+				const groupLength = action.actions.length;
+				let currentElement = { count: 0 };
+
+				action.actions.forEach(groupAction => {
+					let button = this.createButton(groupAction);
+
+					buttons.push({ component: button, toolbarSeparatorAfter: this.toolbarSeparator(groupLength, currentElement) });
+				});
+			}
+		});
+
+		return this.modelView!.modelBuilder.toolbarContainer()
+			.withToolbarItems(
+				buttons
+			).component();
+	}
+
+	private isProjectAction(obj: any): obj is IProjectAction {
+		return obj.id !== undefined;
+	}
+
+	private createButton(projectAction: IProjectAction): azdata.ButtonComponent {
+		let button = this.modelView!.modelBuilder.button()
+			.withProperties<azdata.ButtonProperties>({
+				label: projectAction.id,
+				iconPath: projectAction.icon,
+				height: '20px'
+			}).component();
+
+		button.onDidClick(async () => {
+			await projectAction.run(this.treeItem);
+		});
+
+		return button;
+	}
+
+	/**
+	 * Add a toolbar separator at the end of the group actions
+	 * @param length Total number of actions in a group
+	 * @param currentElement Count of current element
+	 */
+	private toolbarSeparator(length: number, currentElement: { count: number }): boolean {
+		currentElement.count++;
+		if (length === currentElement.count) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	private createContainer(): azdata.FlexContainer {
```

Comment:
Given that this is only used once this doesn't seem that necessary

Reply:
This is just a placeholder for now, It will be properly populated in another PR with data.

Reply:
I'm saying that because it's only called once you can just inline the logic into the method calling it. Having a separate method just to do this doesn't really help much. Typically you should only bother making functions if : 

A. They're going to be called from multiple places
B. The calling function is large/complex enough already

But even for B that really depends on the calling function - the context switch of needing to switch to the function and find out what it's doing can be kind of annoying for someone reading the code.

Reply:
Makes sense. I will get rid of this method.